lightning/src/ln/functional_tests.rs

```diff
@@ -7701,9 +7709,11 @@ fn test_bump_penalty_txn_on_remote_commitment() {
 	let feerate_preimage;
 	{
 		let mut node_txn = nodes[1].tx_broadcaster.txn_broadcasted.lock().unwrap();
-		// 3 transactions including:
+		// 3-6 transactions including:
 		//   preimage and timeout sweeps from remote commitment + preimage sweep bump
-		assert_eq!(node_txn.len(), 3);
+		//   plus, depending on the block connection style, two further bumps
```

Comment:
So the bumping strategy can vary depending on how the user calls the block connection APIs?

How specifically is bumping affected, and could this lead to transactions not confirming in time?